src/test/team_scoreboard.test.ts

```diff
@@ -0,0 +1,296 @@
+import assert from "assert";
+import GuildPreference from "../structures/guild_preference";
+import Team from "../structures/team";
+import TeamScoreboard from "../structures/team_scoreboard";
+import Player from "../structures/player";
+
+const FIRST_TEAM_NAME = "kmq team";
+const SECOND_TEAM_NAME = "not kmqer";
+
+const FIRST_USERID = "12345";
+const TWO_USER_IDS = [FIRST_USERID, "23456"];
+const FOUR_USER_IDS = [FIRST_USERID, "23456", "252525", "1000000"];
+
+const USER_TAG = "unused";
+const AVATAR_URL = "avatarurl";
+
+let scoreboard: TeamScoreboard;
+
+/* eslint-disable prefer-arrow-callback */
+beforeEach(function () {
+    scoreboard = new TeamScoreboard();
+    scoreboard.addTeam(FIRST_TEAM_NAME, new Player("user#0101", FIRST_USERID, AVATAR_URL, 0));
+});
+
+describe("add a team", function () {
+    it("should add the team to the scoreboard", function () {
+        const player = new Player("second_user#1010", TWO_USER_IDS[1], AVATAR_URL, 0);
+        const team = new Team(SECOND_TEAM_NAME, player);
+
+        assert.strictEqual(scoreboard.hasTeam(team.getName()), false);
+        scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        assert.deepStrictEqual(scoreboard.getTeam(team.getName()), team);
+        assert.strictEqual(scoreboard.hasTeam(team.getName()), true);
+        assert.deepStrictEqual(Object.values(scoreboard.getTeams()), [new Team(FIRST_TEAM_NAME, new Player("user#0101", FIRST_USERID, AVATAR_URL, 0)), team]);
+    });
+});
+
+describe("get team of player", function () {
+    it("should get the team that corresponds to the player", function () {
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(FOUR_USER_IDS[0]), scoreboard.getTeam(FIRST_TEAM_NAME));
```

Comment:
Compare with reference to the Team object instead of `scoreboard.getTeam(FIRST_TEAM_NAME)` instead